internal/charmstore/store.go

```diff
@@ -570,9 +570,16 @@ func (s *Store) findEntityInChannel(url *charm.URL, ch params.Channel, fields ma
 	}
 	var entityURL *charm.URL
 	if url.Series == "" {
-		for _, u := range baseEntity.ChannelEntities[ch] {
-			if entityURL == nil || seriesScore[u.Series] > seriesScore[entityURL.Series] {
+		var entitySeries string
+		for s, u := range baseEntity.ChannelEntities[ch] {
+			switch {
```

Comment:
I'm not sure the switch/fallthrough makes things easier to read here.
Maybe:

    if entityURL == nil ||
        u.Series == entityURL.Series && seriesScore[s] > seriesScore[entitySeries] ||
        seriesScore[u.Series] > seriesScore[entityURL.Series] {

?